Unexpected response '404 Not Found' from jira.

Hey. I am really looking forward to use your program but got stuck on the init phase. Is it because of the oldness of my company's jira version or something else?

JiraCLI Version:
```
# jira version
(Version="1.1.0", GitCommit="3b93e147eac468ad985bdece27469153b4bb2814", CommitDate="2022-08-14T08:10:40+00:00", GoVersion="go1.18.3", Compiler="gc", Platform="linux/amd64")
```
Are you using Jira cloud or on-premise jira server? Also mention the version for on-premise installation.
JIRA Service Desk 4.5.7   
Jira Core 8.5.7
![image](https://user-images.githubusercontent.com/4197714/189036306-a568775d-cdbb-4a67-a13f-30e2ecdcaf18.png)
What operating system are you using? Also mention version.
```
# cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

What terminal are you using? Also mention version.
```
# bash --version
GNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)
Copyright (C) 2011 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

```


**To Reproduce**

Steps to reproduce the behavior:


```
# jira init --insecure 
You are using --insecure option. In this mode, the client will NOT verify
server's certificate chain and host name in requests to the jira server.

? Installation type: Local
? Link to Jira server: https://jira.prod
? Login username: test
? Default project: SR

✗ Received unexpected response '404 ' from jira. Please try again.
```

The related issue is: https://github.com/ankitpokhrel/jira-cli/issues/225

@ankitpokhrel  I have test it
```
# docker images
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
REPOSITORY                             TAG         IMAGE ID      CREATED       SIZE
ghcr.io/ankitpokhrel/jira-cli          issue-464   5c8202d7c9c3  41 hours ago  23.8 MB

# jira version
(Version="issue-464", GitCommit="69256adc7de2448516487e889d3235d715f6c9b2", CommitDate="2022-11-19T17:09:00+00:00", GoVersion="go1.17.11", Compiler="gc", Platform="linux/amd64")

```

The error log is
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x7ca417]

goroutine 1 [running]:
net/http/httputil.DumpResponse(0x0, 0x0)
	/usr/local/go/src/net/http/httputil/dump.go:311 +0x37
github.com/ankitpokhrel/jira-cli/pkg/jira.dump(0xc0004c1540, 0xc0002a8700)
	/app/pkg/jira/client.go:260 +0x45
github.com/ankitpokhrel/jira-cli/pkg/jira.(*Client).request.func1()
	/app/pkg/jira/client.go:239 +0x2e
github.com/ankitpokhrel/jira-cli/pkg/jira.(*Client).request(0xc000476380, {0xf0d270, 0xc0000bc000}, {0xce8728, 0x3}, {0xc00029e400, 0x33}, {0x0, 0x0, 0x0}, ...)
	/app/pkg/jira/client.go:255 +0x6a5
github.com/ankitpokhrel/jira-cli/pkg/jira.(*Client).GetV2(...)
	/app/pkg/jira/client.go:167
github.com/ankitpokhrel/jira-cli/pkg/jira.(*Client).Me(0xc000476380)
	/app/pkg/jira/me.go:18 +0xcc
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfigGenerator).verifyLoginDetails(0xc000715cb0, {0xc00029a1b0, 0x21}, {0xc00024bb10, 0xc})
	/app/internal/config/generator.go:293 +0x20f
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfigGenerator).configureServerAndLoginDetails(0xc00055fcb0)
	/app/internal/config/generator.go:277 +0x507
github.com/ankitpokhrel/jira-cli/internal/config.(*JiraCLIConfigGenerator).Generate(0xc000715cb0)
	/app/internal/config/generator.go:117 +0x68
github.com/ankitpokhrel/jira-cli/internal/cmd/init.initialize(0xc0004c3b80, {0xcebe73, 0x2, 0x2})
	/app/internal/cmd/init/init.go:105 +0x205
github.com/spf13/cobra.(*Command).execute(0xc0004c3b80, {0xc000324420, 0x2, 0x2})
	/app/vendor/github.com/spf13/cobra/command.go:876 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004c3900)
	/app/vendor/github.com/spf13/cobra/command.go:990 +0x3bc
main.main()
	/app/cmd/jira/main.go:12 +0x1e
```

--- a/internal/config/generator.go
+++ b/internal/config/generator.go
@@ -329,6 +329,7 @@ func (c *JiraCLIConfigGenerator) configureServerMeta(server, login string) error
 	return nil
 }
 
+//nolint:gocyclo
 func (c *JiraCLIConfigGenerator) configureProjectAndBoardDetails() error {
 	project := c.usrCfg.Project
 	board := c.usrCfg.Board
@@ -405,10 +406,16 @@ func (c *JiraCLIConfigGenerator) configureProjectAndBoardDetails() error {
 	c.value.board = c.boardsMap[strings.ToLower(board)]
 
 	if c.value.board == nil && !strings.EqualFold(board, optionNone) {
+		var suggest string
+		if len(defaultBoardSuggestions) > 2 {
+			suggest = strings.Join(defaultBoardSuggestions[2:], ", ")
+		} else {
+			suggest = strings.Join(defaultBoardSuggestions, ", ")
+		}
 		return fmt.Errorf(
 			"board not found\n  Boards available for the project '%s' are '%s'",
 			c.value.project.Key,
-			strings.Join(defaultBoardSuggestions[2:], ", "),
+			suggest,
 		)
 	}
 	return nil
@@ -638,7 +645,13 @@ func (c *JiraCLIConfigGenerator) getBoardSuggestions(project string) error {
 
 	resp, err := c.jiraClient.Boards(project, "")
 	if err != nil {
-		return err
+		if c.value.installation == jira.InstallationTypeCloud {
+			return err
+		}
+		// We don't care about the error in the local instance since board API may not exist if agile-addon is not installed.
+		// The only option available for board selection, in this case, is "None" if not passed directly from the flag.
+		c.boardSuggestions = append(c.boardSuggestions, optionNone)
+		return nil
 	}
 	c.boardSuggestions = append(c.boardSuggestions, optionSearch, lineBreak)
 	for _, board := range resp.Boards {

--- a/pkg/jira/client.go
+++ b/pkg/jira/client.go
@@ -257,10 +257,12 @@ func (c *Client) request(ctx context.Context, method, endpoint string, body []by
 
 func dump(req *http.Request, res *http.Response) {
 	reqDump, _ := httputil.DumpRequest(req, true)
-	respDump, _ := httputil.DumpResponse(res, false)
-
 	prettyPrintDump("Request Details", reqDump)
-	prettyPrintDump("Response Details", respDump)
+
+	if res != nil {
+		respDump, _ := httputil.DumpResponse(res, false)
+		prettyPrintDump("Response Details", respDump)
+	}
 }
 
 func prettyPrintDump(heading string, data []byte) {

